ui/packages/components/src/Events/EventDetails.tsx

```diff
@@ -0,0 +1,216 @@
+            <ElementWrapper label="Event ID">
+              {isPending ? <SkeletonElement /> : <IDElement>{eventDetailsData?.id}</IDElement>}
+            </ElementWrapper>
+            <ElementWrapper label="Idempotency key">
+              {isPending ? (
+                <SkeletonElement />
+              ) : (
+                <TextElement>{eventDetailsData?.idempotencyKey || '-'}</TextElement>
+              )}
+            </ElementWrapper>
+            <ElementWrapper label="Source">
+              {isPending ? (
+                <SkeletonElement />
+              ) : (
+                <PillElement>{eventDetailsData?.source || 'N/A'}</PillElement>
+              )}
+            </ElementWrapper>
+            <ElementWrapper label="TS">
+              {isPending ? (
+                <SkeletonElement />
+              ) : eventDetailsData?.timestamp ? (
+                <TimeElement date={new Date(eventDetailsData.timestamp)} />
+              ) : (
+                <TextElement>-</TextElement>
+              )}
+            </ElementWrapper>
+            <ElementWrapper label="Version">
+              {isPending ? (
+                <SkeletonElement />
+              ) : (
+                <TextElement>{eventDetailsData?.version || '-'}</TextElement>
+              )}
+            </ElementWrapper>
+          </div>
+          {eventPayloadData?.payload && (
+            <div className="border-subtle border-t pl-px">
+              <CodeBlock
+                header={{ title: 'Payload', ...(error && { status: 'error' }) }}
+                tab={{
+                  content: eventPayloadData?.payload,
+                }}
+                allowFullScreen={true}
+              />
+            </div>
+          )}
+        </div>
+      </div>
+
+      <div className="relative cursor-col-resize" onMouseDown={handleMouseDown}>
+        <div className="bg-canvasMuted absolute inset-0 z-[1] h-full w-px" />
+        <div
+          className="absolute z-[1] -translate-x-1/2"
+          style={{
+            top: (eventInfoRef.current?.clientHeight ?? 0) / 2,
+          }}
+        >
+          <DragDivider className="bg-canvasBase" />
+        </div>
+      </div>
+
+      <div
+        className="border-muted flex flex-col justify-start"
+        style={{ width: `${100 - leftWidth}%` }}
+      >
+        <div className="px-4 py-2">
+          <p className="text-muted mb-4 text-xs font-medium uppercase">Functions Triggered</p>
+          {row.original?.runs?.length ? (
+            <ul className="divide-light divide-y [&>*:not(:first-child)]:pt-[6px] [&>*:not(:last-child)]:pb-[6px]">
+              {row.original.runs.map((run) => (
+                <li key={run.fnSlug}>
+                  <NextLink
+                    href={pathCreator.runPopout({ runID: run.id })}
+                    className="hover:bg-canvasSubtle flex items-center justify-between rounded p-1.5"
+                  >
```

Comment:
This is great - this is going to render so fast if the API has already returned this data on the table!